docs/the-git-based-process-you-need-for-mainframe-development-DRAFT/pipeline-design-and-implementation-supporting-the-workflows.md

```diff
@@ -0,0 +1,326 @@
+>**September 2023 - Release 1.0**
+>
+>**The Git-based process you need for mainframe development**
+>
+>**(Draft Document Sept 1,2023)**  
+
+
+
+**Table of contents**
+
+[Introduction](introduction.md#introduction)
+
+[The Git-based development process for Mainframe development](the-git-based-development-process-for-mainframe-development.md#the-git-based-development-process-for-mainframe-development)
+
+[Pipeline design and implementation supporting the workflows](#pipeline-design-and-implementation-supporting-the-workflows)
+
+1. [Configurations to support working with Feature branches](#configurations-to-support-working-with-feature-branches)
+
+    * [User Build setup](#user-build-setup)
+  
+    * [Pipeline build of feature branches](#pipeline-build-of-feature-branches)
+  
+    * [Package and Deploy a feature for testing in controlled test environments](#package-and-deploy-a-feature-for-testing-in-controlled-test-environments)
+  
+2. [House keeping recommendations](#house-keeping-recommendations)
+
+3. [The Build pipeline for main, epic and release branches](#the-build-pipeline-for-main-epic-and-release-branches)
+
+    * [Build pipeline: Build and Test step](#build-pipeline-build-and-test-step)
+  
+    * [Build pipeline: Install outputs to a shared DEV-TEST runtime](#build-pipeline-install-outputs-to-a-shared-dev-test-runtime)
+  
+    * [Build pipeline: Analyze step](#build-pipeline-analyze-step)
+  
+4. [The Release pipeline: Build, package and deploy](#the-release-pipeline-build-package-and-deploy)
+
+    * [Release pipeline: Build stage](#release-pipeline-build-stage)
+  
+    * [Release pipeline: Packaging stage](#release-pipeline-packaging-stage)
+  
+    * [Release pipeline: Deployment stage](#release-pipeline-deployment-stage)
+  
+5. [Deployment to production](#deployment-to-production)
+
+[Conclusion](conclusion.md#conclusion)
+
+[Appendix](appendix.md#appendix)
+
+
+# Pipeline design and implementation supporting the workflows 
+
+In this section, the technical implementation of the different composition of pipeline of the outlined development methodology will be detailed. If a workflow demands a specific configuration, it will be covered within the same section.
+
+## Configurations to support working with Feature branches
+
+![](images/media/image16.png) When the developers start working on a new task, they will first create a feature branch. Feature branches are created off the latest code state of the state of the source configuration -- the *main*, *epic* or *release* maintenance branch.
+
+If the feature branch was created on the central Git repository, the developers can use the IDE, a terminal, or another Git interface on their local workstation to clone or fetch the new feature branch from the central Git repository down to their local working tree in which the changes will be implemented.
```

Comment:
cloning or fetching isn't sufficient to make the feature branch the one current in the working tree, they will need to `checkout` or `switch` to it.

Reply:
First pass here - https://github.com/IBM/dbb-zappbuild/pull/420